### ISSUE TITLE
Error: Reading phenotype file...

### ISSUE THREAD

**Issue by MuhammadMuneeb007** [shz9/viprs#9]
Reading phenotype file...
Index(['FID', 'IID', 'fatherID', 'motherID', 'sex', 'phenotype',
       'original_index'],
      dtype='object')
Traceback (most recent call last):
  File "/data/ascher01/uqmmune1/miniconda3/envs/viprs_env/lib/python3.11/site-packages/pandas/core/indexes/base.py", line 3803, in get_loc
    return self._engine.get_loc(casted_key)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "pandas/_libs/index.pyx", line 138, in pandas._libs.index.IndexEngine.get_loc
  File "pandas/_libs/index.pyx", line 165, in pandas._libs.index.IndexEngine.get_loc
  File "pandas/_libs/hashtable_class_helper.pxi", line 5745, in pandas._libs.hashtable.PyObjectHashTable.get_item
  File "pandas/_libs/hashtable_class_helper.pxi", line 5753, in pandas._libs.hashtable.PyObjectHashTable.get_item
KeyError: 'phenotype'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/data/ascher01/uqmmune1/BenchmarkingPGSTools/viprs/bin/viprs_fit", line 678, in <module>
    main()
  File "/data/ascher01/uqmmune1/BenchmarkingPGSTools/viprs/bin/viprs_fit", line 615, in main
    data_loaders = init_data(args)
                   ^^^^^^^^^^^^^^^
  File "/data/ascher01/uqmmune1/BenchmarkingPGSTools/viprs/bin/viprs_fit", line 200, in init_data
    validation_gdl = GWADataLoader(
                     ^^^^^^^^^^^^^^
  File "/data/ascher01/uqmmune1/miniconda3/envs/viprs_env/lib/python3.11/site-packages/magenpy/GWADataLoader.py", line 137, in __init__
    self.read_phenotype(phenotype_file)
  File "/data/ascher01/uqmmune1/miniconda3/envs/viprs_env/lib/python3.11/site-packages/magenpy/GWADataLoader.py", line 471, in read_phenotype
    self.sample_table.read_phenotype_file(phenotype_file, drop_na=drop_na, **read_csv_kwargs)
  File "/data/ascher01/uqmmune1/miniconda3/envs/viprs_env/lib/python3.11/site-packages/magenpy/SampleTable.py", line 202, in read_phenotype_file
    if self.table['phenotype'].isnull().all():
       ~~~~~~~~~~^^^^^^^^^^^^^
  File "/data/ascher01/uqmmune1/miniconda3/envs/viprs_env/lib/python3.11/site-packages/pandas/core/frame.py", line 3805, in __getitem__
    indexer = self.columns.get_loc(key)
              ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/data/ascher01/uqmmune1/miniconda3/envs/viprs_env/lib/python3.11/site-packages/pandas/core/indexes/base.py", line 3805, in get_loc
    raise KeyError(key) from err
KeyError: 'phenotype'




Solution

![image](https://github.com/user-attachments/assets/e4b0e04d-c368-4147-9119-965a67df85c6)
file path /envs/viprs_env/lib/python3.11/site-packages/magenpy/SampleTable.py

**Comment by shz9** [comment]
Thanks for catching this bug Muhammad!
I indeed fixed this issue in `magenpy` recently, but haven't pushed the latest changes to GitHub/PyPI yet. Will do so soon.

**Comment by shz9** [comment]
This should now be fixed as part of `magenpy==0.1.4`.
